handlers.go

```diff
@@ -190,6 +212,42 @@ func (dht *IpfsDHT) handlePutValue(ctx context.Context, p peer.ID, pmes *pb.Mess
 	return pmes, err
 }
 
+func (dht *IpfsDHT) getRecordFromDatastore(dskey ds.Key) (*recpb.Record, error) {
```

Comment:
Would it be possible to dedup some of this code with `checkLocalDatastore`?

Reply:
It's a little tricky because for most of the errors in `checkLocalDatastore` we want to return an error, whereas for most of the errors in `getRecordFromDatastore` we want to log a message and ignore them (because we can just overwrite a bad record)

Reply:
Also we really want to return that `nil` so that such errors in existing record checking don't prohibit update.